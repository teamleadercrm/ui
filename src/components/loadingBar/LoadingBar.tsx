import cx from 'classnames';
import React from 'react';
import { GenericComponent } from '../../@types/types';
import Box from '../box';
import { BoxProps } from '../box/Box';
import theme from './theme.css';

export interface LoadingBarProps extends Omit<BoxProps, 'className' | 'size'> {
  /** A class name for the wrapper to add custom classes */
  className?: string;
  /** The color of the components */
  color?: 'aqua' | 'gold' | 'mint' | 'neutral' | 'ruby' | 'teal' | 'violet';
  /** Size of the component */
  size?: 'small' | 'medium' | 'large';
  /** The tint of the components color */
  tint?: 'lightest' | 'light' | 'normal' | 'dark' | 'darkest';
}

const LoadingBar: GenericComponent<LoadingBarProps> = ({
  className,
  color = 'mint',
  size = 'small',
  tint = 'normal',
  ...others
}) => {
  const classNames = cx(
    theme['loading-bar'],
    theme[`is-${color}`],
    theme[`is-${size}`],
    theme[`is-${tint}`],
    className,
  );
  return (
    <Box data-teamleader-ui="loading-bar" className={classNames} {...others}>
      <div className={theme['loading-bar-indicator']} />
    </Box>
  );
};

export default LoadingBar;
