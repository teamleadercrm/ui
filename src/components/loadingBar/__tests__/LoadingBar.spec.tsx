import React from 'react';
import { render } from '@testing-library/react';
import LoadingBar from '..';

describe('Component - LoadingBar', () => {
  it('renders', async () => {
    const { asFragment } = render(<LoadingBar />);
    expect(asFragment()).toMatchSnapshot();
  });
});
