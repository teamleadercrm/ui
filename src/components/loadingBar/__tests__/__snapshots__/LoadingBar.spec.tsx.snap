// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component - LoadingBar renders 1`] = `
<DocumentFragment>
  <div
    class="box loading-bar is-mint is-small is-normal"
    data-teamleader-ui="loading-bar"
  >
    <div
      class="loading-bar-indicator"
    />
  </div>
</DocumentFragment>
`;
